src/symbolic/scoped_symbol.ml

```diff
@@ -0,0 +1,107 @@
+type scope_token =
```

Comment:
Maybe rename the whole module from `scoped_symbol` to `symbol_scope` (because we let the user work with scopes now, and not only the symbol as it was done in the beginning).

Then this type could be simply `token`